cla_backend/apps/legalaid/views.py

```diff
@@ -177,7 +178,10 @@ def create_means_test_log(self, obj, created):
         except Case.DoesNotExist:
             return
 
-        user = self.request.user
+        if self.request.user:
```

Comment:
I would personally implement a `get_user` method on the `legalaid.views.BaseEligibilityCheckViewSet` which defaults to `return self.request.user` and then override that method in `checker.views. EligibilityCheckViewSet` to return `get_web_user()`.

This way, we could catch potential errors when the user is not authenticated but it should be (e.g. in call_centre.views.EligibilityCheckViewSet)